packages/cra-template/template/src/Components/Pager.tsx

```diff
@@ -14,7 +14,10 @@ const PagerRenderer: FunctionComponent<PagerProps> = (props) => {
   const [state, setState] = useState(controller.state);
 
   useEffect(
-    () => controller.subscribe(() => setState(controller.state)),
+    () =>
+      controller.subscribe(() =>
+        process.nextTick(() => setState(controller.state))
+      ),
```

Comment:
This is fixing the following error:  
```
Cannot update a component ("PagerRenderer") while rendering a different component ("ResultListRenderer")
```

In all the documentation I found, it says that the error can be prevented by using enclosing the `setState` method inside the `useEffect` hook (which is already the case here).
So, is it possible that the Pager and ResultList controllers are "dispatching" their new state at the same time, which causes the 2 components to trigger a page render at the same moment?

Reply:
My React is rusty but... process.nextTick ? in a browser ?

Reply:
Oh you're right. I'll switch to setTimeout(0).

Reply:
> Well, it is supported but I'll switch to setTimeout(0).

But... How 😕 When you break on this line, what is going on 🤯 what is process?

Reply:
It's an implementation of process for browsers
https://www.npmjs.com/package/process

Reply:
Mh why not then, curious where does it come from?